vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+
+## Derive `NADIR` {#nadir}
+
+Once we have a TSD parameter, the NADIR is set to the lowest sum of diameters in
+Analysis Value before the current observation, where Analysis Flag 01 =Y.  The
+first observations after baseline (depending on how you have defined baseline
+earlier) is set to analysis value where baseline record flag = Y
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_vars_joined(
+    dataset_add = adtr_sum,
+    by_vars = vars(USUBJID),
+    order = vars(AVAL),
+    new_vars = vars(NADIR = AVAL),
+    join_vars = vars(ADY),
+    filter_add = ADY > 0 & ANL01FL == "Y",
+    filter_join = ADY.join < ADY,
+    mode = "first",
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, PARAMCD, AVISIT, AVAL, NADIR)
+)
+```  
+
+## Derive Change from Baseline (`CHG`, `PCHG`) {#chg}
+
+These functions are available in admiral, below examples show its usage for
+`ADTR`.  We calculate the change from baseline and percentage change from
+baseline for the parameter TSD and the variable `NADIR`.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_chg() %>%
+  derive_var_pchg() %>%
+  mutate(
+    CHGNAD = AVAL - NADIR,
+    PCHGNAD = 100 * CHGNAD / NADIR
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, CHG, CHGNAD, PCHGNAD)
+)
+```  
+
+## Derive Additional Flag Variables {#addflag}
+
+Depending on the analysis flags you require, you may want to create additional
+flag variables that will then aid setting up any additional analysis flag
+variables defined as per your company specifications.
+
+###  Derive `CRFL`, `CRNFL`
+
+In this example, `CRFL` flags where the patient has had Complete Response, and
+`CRNFL`, flags where Complete response contributed to the NADIR calculation.
+
+```{r}
+adtr_sum <- adtr_sum %>% mutate(
+  CRFL = if_else(AVAL == 0 & ANL01FL == "Y", "Y", NA_character_),
+  CRNFL = if_else(NADIR == 0, "Y", NA_character_)
+)
+```
+
+### Derive `PDFL`
+
+In this example, we want to flag when a patient has had PD.  This could be done
+in a number of ways:
+
+1. take the date of PD from `ADRS`
+
+2. take the first date when the overall response is PD from `RS` or
+
+3. Calculate from source data, the derivation for PD for target lesions is based
+on the calculated sum of diameters of the assessed lesions results in a
+calculated Percent Change from NADIR >=20% and a calculated Change from NADIR
+>=5mm depending on your company specifications.
+
+See below example.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  mutate(
+    PDFL = if_else(PCHGNAD >= 20 & CHGNAD >= 5 | is.na(CRFL) & CRNFL == "Y", "Y", NA_character_)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, CHGNAD, PCHGNAD, PDFL)
+)
+```  
+
+## Derive Analysis Flags `ANL02FL`, `ANL03FL`, `ANL04FL` {#analflag}
```

Comment:
Should we replace "`ANL02FL`, `ANL03FL`, `ANL04FL`" with "(`ANLzzFL`)"?

Reply:
Also can we structure this section a little, maybe with sub-sections, just for readability purposes.

Reply:
done